File: web/src/types/branch.ts
// Branch types - mirrors internal/domain/branch.go

export interface Branch {
  id: number
  name: string
  code: string
  address?: string
  phone?: string
  email?: string
  is_active: boolean
  timezone: string
  currency: string
  default_interest_rate: number
  default_loan_term_days: number
  default_grace_period: number
  created_at: string
  updated_at: string
  deleted_at?: string
}

export interface CreateBranchInput {
  name: string
  code: string
  address?: string
  phone?: string
  email?: string
  timezone?: string
  currency?: string
  default_interest_rate?: number
  default_loan_term_days?: number
  default_grace_period?: number
}

export interface UpdateBranchInput {
  name?: string
  address?: string
  phone?: string
  email?: string
  timezone?: string
  currency?: string
  is_active?: boolean
  default_interest_rate?: number
  default_loan_term_days?: number
  default_grace_period?: number
}

File: internal/domain/branch.go
package domain

import (
	"time"
)

// Branch represents a pawnshop branch/location
type Branch struct {
	ID        int64  `json:"id"`
	Name      string `json:"name"`
	Code      string `json:"code"`
	Address   string `json:"address,omitempty"`
	Phone     string `json:"phone,omitempty"`
	Email     string `json:"email,omitempty"`
	IsActive  bool   `json:"is_active"`
	Timezone  string `json:"timezone"`
	Currency  string `json:"currency"`

	// Business settings
	DefaultInterestRate float64 `json:"default_interest_rate"`
	DefaultLoanTermDays int     `json:"default_loan_term_days"`
	DefaultGracePeriod  int     `json:"default_grace_period"`

	// Timestamps
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at,omitempty"`
}

// TableName returns the database table name
func (Branch) TableName() string {
	return "branches"
}
